vocabulary-care-pathways/api/src/main/java/org/phenotips/vocabulary/internal/solr/CPNode.java

```diff
@@ -0,0 +1,193 @@
+
+/**
+ * A class that holds data pertaining to a care pathways test or care node. Has convenience methods such as
+ * {@link #addAncestor(String)}, {@link #addAncestors(Collection)}, {@link #setAncestors(Set)}, {@link #getAncestors()},
+ * etc. for easy data access.
+ */
+class CPNode
+{
+    /** The node name. */
+    private final String name;
+
+    /** The node id. */
+    private final String id;
+
+    /** The set of parent identifiers for the node. */
+    private Set<String> isA;
+
+    /** The set of ancestor identifiers for the node. */
+    private Set<String> termCategory;
+
+    /**
+     * A simple constructor, requiring the node {@code id} and {@code name}.
+     *
+     * @param id the node identifier, as string; must not be blank
+     * @param name the node name, as string; must not be blank
+     */
+    CPNode(@Nonnull final String id, @Nonnull final String name)
+    {
+        this.id = id;
+        this.name = name;
+        this.isA = new HashSet<>();
+        this.termCategory = new HashSet<>();
+    }
+
+    /**
+     * Gets the node {@link #id}.
+     *
+     * @return the node identifier
+     */
+    @Nonnull
+    String getId()
+    {
+        return this.id;
+    }
+
+    /**
+     * Gets the node {@link #name}.
+     *
+     * @return the node name
+     */
+    @Nonnull
+    String getName()
+    {
+        return this.name;
+    }
+
+    /**
+     * Returns the collection of parent identifiers for the node.
+     *
+     * @return a collection of parents
+     */
+    @Nonnull
+    Collection<String> getParents()
+    {
+        return Collections.unmodifiableSet(this.isA);
+    }
+
+    /**
+     * Returns the collection of ancestor identifiers for the node.
+     *
+     * @return a collection of identifiers
+     */
+    @Nonnull
+    Collection<String> getAncestors()
+    {
+        return Collections.unmodifiableSet(this.termCategory);
+    }
+
+    /**
+     * Adds a parent to the collection of {@link #isA parents}, and returns the updated collection.
+     *
+     * @param parent the parent identifier to add
+     * @return the updated collection of parent identifiers
+     */
+    @Nonnull
+    Collection<String> addParent(@Nullable final String parent)
+    {
+        CollectionUtils.addIgnoreNull(this.isA, parent);
+        return Collections.unmodifiableSet(this.isA);
+    }
+
+    /**
+     * Adds an ancestor to the collection of {@link #termCategory ancestors}, and returns the updated collection.
+     *
+     * @param ancestor the ancestor identifier to add
+     * @return the updated collection of ancestor identifiers
+     */
+    @Nonnull
+    Collection<String> addAncestor(@Nullable final String ancestor)
+    {
+        CollectionUtils.addIgnoreNull(this.termCategory, ancestor);
+        return Collections.unmodifiableSet(this.termCategory);
+    }
+
+    /**
+     * Adds a parents to the collection of {@link #isA parents}, and returns the updated collection.
```

Comment:
`Adds a parents` isn't correct, perhaps `Adds a collection of parents` or `Adds several parents at once`, or as you said for `addAncestors`, just `Adds parents`?